truffle/src/main/java/org/jruby/truffle/options/OptionTypeException.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.options;
+
+public class OptionTypeException extends UnsupportedOperationException {
```

Comment:
I think it'd be helpful for debugging if this exception type took the option name as its argument. Or perhaps we'd just enforce a message passed with the option name.